From 6bab1aae419bb711479d00b8932ea48e4cbe4c5a Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 14 Feb 2025 15:32:26 +1100
Subject: [PATCH 6/6] tell mkl about the right xerbla

automated using the following script
```
import re
import os

pat_external = re.compile(r"(?m)(EXTERNAL[\w\s\n,\$]*\bXERBLA)(?!, MKL)")
pat_invoke = re.compile(r"(?m)^(\*[\s\.]*Executable Statements[\s\.\n\*]*)")
invocation = """\
*
      CALL MKL_INIT_XERBLA()
"""

def rewrite(path):
    with open(path, "r+") as f:
        lines = f.readlines()
        # add MKL_INIT_XERBLA to declaration of external subroutines;
        # use subn to track whether any replacements happened
        replaced = [pat_external.subn(r"\1, MKL_INIT_XERBLA", x) for x in lines]
        lines, num_replaced = zip(*replaced)
        if not any(num_replaced):
            return
        # still need to call MKL_INIT_XERBLA after we've added it to the external declarations
        lines = [pat_invoke.sub(r"\1" + invocation, x) for x in lines]
        # overwrite file
        f.seek(0)
        f.write("".join(lines))
        f.truncate()

for folder, _, files in os.walk("./TESTING"):
    for file in [f for f in files if f.endswith(".f")]:
        rewrite(os.path.join(folder, file))
```
---
 TESTING/EIG/cchkhb2stg.f  | 4 +++-
 TESTING/EIG/cdrgvx.f      | 4 +++-
 TESTING/EIG/cdrvbd.f      | 4 +++-
 TESTING/EIG/cdrvsg2stg.f  | 4 +++-
 TESTING/EIG/cdrvst2stg.f  | 4 +++-
 TESTING/EIG/cget23.f      | 4 +++-
 TESTING/EIG/cget24.f      | 4 +++-
 TESTING/EIG/csbmv.f       | 4 +++-
 TESTING/EIG/cunt03.f      | 4 +++-
 TESTING/EIG/ddrgvx.f      | 4 +++-
 TESTING/EIG/dget23.f      | 4 +++-
 TESTING/EIG/dget24.f      | 4 +++-
 TESTING/EIG/dort03.f      | 4 +++-
 TESTING/EIG/sdrgvx.f      | 4 +++-
 TESTING/EIG/sget23.f      | 4 +++-
 TESTING/EIG/sget24.f      | 4 +++-
 TESTING/EIG/sort03.f      | 4 +++-
 TESTING/EIG/zchkbb.f      | 4 +++-
 TESTING/EIG/zchkbd.f      | 4 +++-
 TESTING/EIG/zchkgg.f      | 4 +++-
 TESTING/EIG/zchkhb.f      | 4 +++-
 TESTING/EIG/zchkhb2stg.f  | 4 +++-
 TESTING/EIG/zchkhs.f      | 4 +++-
 TESTING/EIG/zdrges.f      | 4 +++-
 TESTING/EIG/zdrges3.f     | 4 +++-
 TESTING/EIG/zdrgev.f      | 4 +++-
 TESTING/EIG/zdrgev3.f     | 4 +++-
 TESTING/EIG/zdrgsx.f      | 4 +++-
 TESTING/EIG/zdrgvx.f      | 4 +++-
 TESTING/EIG/zdrvbd.f      | 4 +++-
 TESTING/EIG/zdrves.f      | 4 +++-
 TESTING/EIG/zdrvev.f      | 4 +++-
 TESTING/EIG/zdrvsg.f      | 4 +++-
 TESTING/EIG/zdrvsg2stg.f  | 4 +++-
 TESTING/EIG/zdrvst.f      | 4 +++-
 TESTING/EIG/zdrvst2stg.f  | 4 +++-
 TESTING/EIG/zdrvsx.f      | 4 +++-
 TESTING/EIG/zdrvvx.f      | 4 +++-
 TESTING/EIG/zget23.f      | 4 +++-
 TESTING/EIG/zget24.f      | 4 +++-
 TESTING/EIG/zlarhs.f      | 4 +++-
 TESTING/EIG/zsbmv.f       | 4 +++-
 TESTING/EIG/zunt03.f      | 4 +++-
 TESTING/LIN/cgelqs.f      | 4 +++-
 TESTING/LIN/cgeqls.f      | 4 +++-
 TESTING/LIN/cgeqrs.f      | 4 +++-
 TESTING/LIN/cgerqs.f      | 4 +++-
 TESTING/LIN/clavhe.f      | 4 +++-
 TESTING/LIN/clavhe_rook.f | 4 +++-
 TESTING/LIN/clavhp.f      | 4 +++-
 TESTING/LIN/clavsp.f      | 4 +++-
 TESTING/LIN/clavsy.f      | 4 +++-
 TESTING/LIN/clavsy_rook.f | 4 +++-
 TESTING/LIN/cqpt01.f      | 4 +++-
 TESTING/LIN/cqrt11.f      | 4 +++-
 TESTING/LIN/cqrt14.f      | 4 +++-
 TESTING/LIN/cqrt17.f      | 4 +++-
 TESTING/LIN/crzt01.f      | 4 +++-
 TESTING/LIN/crzt02.f      | 4 +++-
 TESTING/LIN/csbmv.f       | 4 +++-
 TESTING/LIN/dgelqs.f      | 4 +++-
 TESTING/LIN/dgeqls.f      | 4 +++-
 TESTING/LIN/dgeqrs.f      | 4 +++-
 TESTING/LIN/dgerqs.f      | 4 +++-
 TESTING/LIN/dlavsp.f      | 4 +++-
 TESTING/LIN/dlavsy.f      | 4 +++-
 TESTING/LIN/dlavsy_rook.f | 4 +++-
 TESTING/LIN/dqpt01.f      | 4 +++-
 TESTING/LIN/dqrt11.f      | 4 +++-
 TESTING/LIN/dqrt14.f      | 4 +++-
 TESTING/LIN/dqrt17.f      | 4 +++-
 TESTING/LIN/drzt01.f      | 4 +++-
 TESTING/LIN/drzt02.f      | 4 +++-
 TESTING/LIN/dtplqt.f      | 4 +++-
 TESTING/LIN/sgelqs.f      | 4 +++-
 TESTING/LIN/sgeqls.f      | 4 +++-
 TESTING/LIN/sgeqrs.f      | 4 +++-
 TESTING/LIN/sgerqs.f      | 4 +++-
 TESTING/LIN/slavsp.f      | 4 +++-
 TESTING/LIN/slavsy.f      | 4 +++-
 TESTING/LIN/slavsy_rook.f | 4 +++-
 TESTING/LIN/sqpt01.f      | 4 +++-
 TESTING/LIN/sqrt11.f      | 4 +++-
 TESTING/LIN/sqrt14.f      | 4 +++-
 TESTING/LIN/sqrt17.f      | 4 +++-
 TESTING/LIN/srzt01.f      | 4 +++-
 TESTING/LIN/srzt02.f      | 4 +++-
 TESTING/LIN/stplqt.f      | 4 +++-
 TESTING/LIN/zgelqs.f      | 4 +++-
 TESTING/LIN/zgeqls.f      | 4 +++-
 TESTING/LIN/zgeqrs.f      | 4 +++-
 TESTING/LIN/zgerqs.f      | 4 +++-
 TESTING/LIN/zlarhs.f      | 4 +++-
 TESTING/LIN/zlavhe.f      | 4 +++-
 TESTING/LIN/zlavhe_rook.f | 4 +++-
 TESTING/LIN/zlavhp.f      | 4 +++-
 TESTING/LIN/zlavsp.f      | 4 +++-
 TESTING/LIN/zlavsy.f      | 4 +++-
 TESTING/LIN/zlavsy_rook.f | 4 +++-
 TESTING/LIN/zqpt01.f      | 4 +++-
 TESTING/LIN/zqrt11.f      | 4 +++-
 TESTING/LIN/zqrt12.f      | 4 +++-
 TESTING/LIN/zqrt14.f      | 4 +++-
 TESTING/LIN/zqrt15.f      | 4 +++-
 TESTING/LIN/zqrt17.f      | 4 +++-
 TESTING/LIN/zrzt01.f      | 4 +++-
 TESTING/LIN/zrzt02.f      | 4 +++-
 TESTING/LIN/zsbmv.f       | 4 +++-
 TESTING/MATGEN/clagge.f   | 4 +++-
 TESTING/MATGEN/clahilb.f  | 4 +++-
 TESTING/MATGEN/clarge.f   | 4 +++-
 TESTING/MATGEN/claror.f   | 4 +++-
 TESTING/MATGEN/clarot.f   | 4 +++-
 TESTING/MATGEN/clatm1.f   | 4 +++-
 TESTING/MATGEN/clatmr.f   | 4 +++-
 TESTING/MATGEN/dlagge.f   | 4 +++-
 TESTING/MATGEN/dlahilb.f  | 4 +++-
 TESTING/MATGEN/dlarge.f   | 4 +++-
 TESTING/MATGEN/dlaror.f   | 4 +++-
 TESTING/MATGEN/dlarot.f   | 4 +++-
 TESTING/MATGEN/dlatm1.f   | 4 +++-
 TESTING/MATGEN/dlatm7.f   | 4 +++-
 TESTING/MATGEN/dlatmr.f   | 4 +++-
 TESTING/MATGEN/slagge.f   | 4 +++-
 TESTING/MATGEN/slahilb.f  | 4 +++-
 TESTING/MATGEN/slarge.f   | 4 +++-
 TESTING/MATGEN/slaror.f   | 4 +++-
 TESTING/MATGEN/slarot.f   | 4 +++-
 TESTING/MATGEN/slatm1.f   | 4 +++-
 TESTING/MATGEN/slatm7.f   | 4 +++-
 TESTING/MATGEN/slatmr.f   | 4 +++-
 TESTING/MATGEN/zlagge.f   | 4 +++-
 TESTING/MATGEN/zlaghe.f   | 4 +++-
 TESTING/MATGEN/zlagsy.f   | 4 +++-
 TESTING/MATGEN/zlahilb.f  | 4 +++-
 TESTING/MATGEN/zlarge.f   | 4 +++-
 TESTING/MATGEN/zlaror.f   | 4 +++-
 TESTING/MATGEN/zlarot.f   | 4 +++-
 TESTING/MATGEN/zlatm1.f   | 4 +++-
 TESTING/MATGEN/zlatme.f   | 4 +++-
 TESTING/MATGEN/zlatmr.f   | 4 +++-
 TESTING/MATGEN/zlatms.f   | 4 +++-
 TESTING/MATGEN/zlatmt.f   | 4 +++-
 143 files changed, 429 insertions(+), 143 deletions(-)

diff --git a/TESTING/EIG/cchkhb2stg.f b/TESTING/EIG/cchkhb2stg.f
index 61537f44b..fdc9e40d5 100644
--- a/TESTING/EIG/cchkhb2stg.f
+++ b/TESTING/EIG/cchkhb2stg.f
@@ -372,7 +372,7 @@
       EXTERNAL           SLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SLASUM, XERBLA, CHBT21, CHBTRD, CLACPY, CLASET,
+      EXTERNAL           SLASUM, XERBLA, MKL_INIT_XERBLA, CHBT21, CHBTRD, CLACPY, CLASET,
      $                   CLATMR, CLATMS, CHETRD_HB2ST, CSTEQR
 *     ..
 *     .. Intrinsic Functions ..
@@ -386,6 +386,8 @@
      $                   0, 0 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/cdrgvx.f b/TESTING/EIG/cdrgvx.f
index bed8e472f..ccaae40c7 100644
--- a/TESTING/EIG/cdrgvx.f
+++ b/TESTING/EIG/cdrgvx.f
@@ -341,12 +341,14 @@
       EXTERNAL           ILAENV, CLANGE, SLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, CGET52, CGGEVX, CLACPY, CLATM6, XERBLA
+      EXTERNAL           ALASVM, CGET52, CGGEVX, CLACPY, CLATM6, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, CMPLX, MAX, SQRT
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/cdrvbd.f b/TESTING/EIG/cdrvbd.f
index 7b7b01b47..6b15c4125 100644
--- a/TESTING/EIG/cdrvbd.f
+++ b/TESTING/EIG/cdrvbd.f
@@ -458,7 +458,7 @@
       EXTERNAL           SLAMCH, SLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, XERBLA, CBDT01, CBDT05, CGESDD,
+      EXTERNAL           ALASVM, XERBLA, MKL_INIT_XERBLA, CBDT01, CBDT05, CGESDD,
      $                   CGESVD, CGESVDQ, CGESVJ, CGEJSV, CGESVDX,
      $                   CLACPY, CLASET, CLATMS, CUNT01, CUNT03
 *     ..
@@ -477,6 +477,8 @@
       DATA               CJOBV / 'N', 'V' /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/cdrvsg2stg.f b/TESTING/EIG/cdrvsg2stg.f
index 3b7156b42..57642acc8 100644
--- a/TESTING/EIG/cdrvsg2stg.f
+++ b/TESTING/EIG/cdrvsg2stg.f
@@ -429,7 +429,7 @@
       EXTERNAL           LSAME, SLAMCH, SLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SLABAD, SLAFTS, SLASUM, XERBLA, CHBGV, CHBGVD,
+      EXTERNAL           SLABAD, SLAFTS, SLASUM, XERBLA, MKL_INIT_XERBLA, CHBGV, CHBGVD,
      $                   CHBGVX, CHEGV, CHEGVD, CHEGVX, CHPGV, CHPGVD,
      $                   CHPGVX, CLACPY, CLASET, CLATMR, CLATMS, CSGT01,
      $                   CHEGV_2STAGE
@@ -445,6 +445,8 @@
      $                   0, 0, 6*4 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Check for errors
 *
diff --git a/TESTING/EIG/cdrvst2stg.f b/TESTING/EIG/cdrvst2stg.f
index 095bd7885..a9ce386f8 100644
--- a/TESTING/EIG/cdrvst2stg.f
+++ b/TESTING/EIG/cdrvst2stg.f
@@ -394,7 +394,7 @@
       EXTERNAL           SLAMCH, SLARND, SSXT1
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, SLABAD, SLAFTS, XERBLA, CHBEV, CHBEVD,
+      EXTERNAL           ALASVM, SLABAD, SLAFTS, XERBLA, MKL_INIT_XERBLA, CHBEV, CHBEVD,
      $                   CHBEVX, CHEEV, CHEEVD, CHEEVR, CHEEVX, CHET21,
      $                   CHET22, CHPEV, CHPEVD, CHPEVX, CLACPY, CLASET,
      $                   CHEEVD_2STAGE, CHEEVR_2STAGE, CHEEVX_2STAGE,
@@ -412,6 +412,8 @@
      $                   0, 0, 4, 4, 4 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Check for errors
 *
diff --git a/TESTING/EIG/cget23.f b/TESTING/EIG/cget23.f
index 1a2d4fb48..31e702779 100644
--- a/TESTING/EIG/cget23.f
+++ b/TESTING/EIG/cget23.f
@@ -420,7 +420,7 @@
       EXTERNAL           LSAME, SCNRM2, SLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGEEVX, CGET22, CLACPY, XERBLA
+      EXTERNAL           CGEEVX, CGET22, CLACPY, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, AIMAG, MAX, MIN, REAL
@@ -429,6 +429,8 @@
       DATA               SENS / 'N', 'V' /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/cget24.f b/TESTING/EIG/cget24.f
index e97003a2f..2f21afed7 100644
--- a/TESTING/EIG/cget24.f
+++ b/TESTING/EIG/cget24.f
@@ -384,7 +384,7 @@
       EXTERNAL           CSLECT, CLANGE, SLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CCOPY, CGEESX, CGEMM, CLACPY, CUNT01, XERBLA
+      EXTERNAL           CCOPY, CGEESX, CGEMM, CLACPY, CUNT01, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, AIMAG, MAX, MIN, REAL
@@ -400,6 +400,8 @@
       COMMON             / SSLCT / SELOPT, SELDIM, SELVAL, SELWR, SELWI
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/csbmv.f b/TESTING/EIG/csbmv.f
index dc1b2330d..4d1926058 100644
--- a/TESTING/EIG/csbmv.f
+++ b/TESTING/EIG/csbmv.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, MIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/EIG/cunt03.f b/TESTING/EIG/cunt03.f
index 8e1fd207f..6ef83ae6c 100644
--- a/TESTING/EIG/cunt03.f
+++ b/TESTING/EIG/cunt03.f
@@ -199,9 +199,11 @@
       INTRINSIC          ABS, CMPLX, MAX, MIN, REAL
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CUNT01, XERBLA
+      EXTERNAL           CUNT01, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check inputs
 *
diff --git a/TESTING/EIG/ddrgvx.f b/TESTING/EIG/ddrgvx.f
index 32d08b5d7..cffb8e03f 100644
--- a/TESTING/EIG/ddrgvx.f
+++ b/TESTING/EIG/ddrgvx.f
@@ -342,12 +342,14 @@
       EXTERNAL           ILAENV, DLAMCH, DLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, DGET52, DGGEVX, DLACPY, DLATM6, XERBLA
+      EXTERNAL           ALASVM, DGET52, DGGEVX, DLACPY, DLATM6, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX, SQRT
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/dget23.f b/TESTING/EIG/dget23.f
index 8c801aeee..326b4e64f 100644
--- a/TESTING/EIG/dget23.f
+++ b/TESTING/EIG/dget23.f
@@ -428,7 +428,7 @@
       EXTERNAL           LSAME, DLAMCH, DLAPY2, DNRM2
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DGEEVX, DGET22, DLACPY, XERBLA
+      EXTERNAL           DGEEVX, DGET22, DLACPY, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, MAX, MIN
@@ -437,6 +437,8 @@
       DATA               SENS / 'N', 'V' /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/dget24.f b/TESTING/EIG/dget24.f
index a3c01756c..800ebf48b 100644
--- a/TESTING/EIG/dget24.f
+++ b/TESTING/EIG/dget24.f
@@ -397,12 +397,14 @@
       EXTERNAL           DSLECT, DLAMCH, DLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DCOPY, DGEESX, DGEMM, DLACPY, DORT01, XERBLA
+      EXTERNAL           DCOPY, DGEESX, DGEMM, DLACPY, DORT01, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, MAX, MIN, SIGN, SQRT
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/dort03.f b/TESTING/EIG/dort03.f
index 46f34574f..9388beb32 100644
--- a/TESTING/EIG/dort03.f
+++ b/TESTING/EIG/dort03.f
@@ -190,9 +190,11 @@
       INTRINSIC          ABS, DBLE, MAX, MIN, SIGN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DORT01, XERBLA
+      EXTERNAL           DORT01, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check inputs
 *
diff --git a/TESTING/EIG/sdrgvx.f b/TESTING/EIG/sdrgvx.f
index b6cc91dc4..c9b1cf0dc 100644
--- a/TESTING/EIG/sdrgvx.f
+++ b/TESTING/EIG/sdrgvx.f
@@ -344,12 +344,14 @@
       EXTERNAL           ILAENV, SLAMCH, SLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, SGET52, SGGEVX, SLACPY, SLATM6, XERBLA
+      EXTERNAL           ALASVM, SGET52, SGGEVX, SLACPY, SLATM6, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX, SQRT
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/sget23.f b/TESTING/EIG/sget23.f
index ac936bc89..60d1cc752 100644
--- a/TESTING/EIG/sget23.f
+++ b/TESTING/EIG/sget23.f
@@ -428,7 +428,7 @@
       EXTERNAL           LSAME, SLAMCH, SLAPY2, SNRM2
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SGEEVX, SGET22, SLACPY, XERBLA
+      EXTERNAL           SGEEVX, SGET22, SLACPY, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX, MIN, REAL
@@ -437,6 +437,8 @@
       DATA               SENS / 'N', 'V' /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/sget24.f b/TESTING/EIG/sget24.f
index e0b75ff0f..7b5951acd 100644
--- a/TESTING/EIG/sget24.f
+++ b/TESTING/EIG/sget24.f
@@ -397,12 +397,14 @@
       EXTERNAL           SSLECT, SLAMCH, SLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SCOPY, SGEESX, SGEMM, SLACPY, SORT01, XERBLA
+      EXTERNAL           SCOPY, SGEESX, SGEMM, SLACPY, SORT01, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX, MIN, REAL, SIGN, SQRT
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/sort03.f b/TESTING/EIG/sort03.f
index 09a941487..31d6f3a58 100644
--- a/TESTING/EIG/sort03.f
+++ b/TESTING/EIG/sort03.f
@@ -190,9 +190,11 @@
       INTRINSIC          ABS, MAX, MIN, REAL, SIGN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SORT01, XERBLA
+      EXTERNAL           SORT01, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check inputs
 *
diff --git a/TESTING/EIG/zchkbb.f b/TESTING/EIG/zchkbb.f
index 877caffa5..adb409999 100644
--- a/TESTING/EIG/zchkbb.f
+++ b/TESTING/EIG/zchkbb.f
@@ -409,7 +409,7 @@
       EXTERNAL           DLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLAHD2, DLASUM, XERBLA, ZBDT01, ZBDT02, ZGBBRD,
+      EXTERNAL           DLAHD2, DLASUM, XERBLA, MKL_INIT_XERBLA, ZBDT01, ZBDT02, ZGBBRD,
      $                   ZLACPY, ZLASET, ZLATMR, ZLATMS, ZUNT01
 *     ..
 *     .. Intrinsic Functions ..
@@ -422,6 +422,8 @@
      $                   0, 0 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zchkbd.f b/TESTING/EIG/zchkbd.f
index b3d560554..805c9d136 100644
--- a/TESTING/EIG/zchkbd.f
+++ b/TESTING/EIG/zchkbd.f
@@ -466,7 +466,7 @@
       EXTERNAL           DLAMCH, DLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASUM, DCOPY, DLABAD, DLAHD2, DSVDCH, XERBLA,
+      EXTERNAL           ALASUM, DCOPY, DLABAD, DLAHD2, DSVDCH, XERBLA, MKL_INIT_XERBLA,
      $                   ZBDSQR, ZBDT01, ZBDT02, ZBDT03, ZGEBRD, ZGEMM,
      $                   ZLACPY, ZLASET, ZLATMR, ZLATMS, ZUNGBR, ZUNT01
 *     ..
@@ -489,6 +489,8 @@
      $                   0, 0, 0 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zchkgg.f b/TESTING/EIG/zchkgg.f
index 2f99d76f3..c769a7ecd 100644
--- a/TESTING/EIG/zchkgg.f
+++ b/TESTING/EIG/zchkgg.f
@@ -562,7 +562,7 @@
       EXTERNAL           DLAMCH, ZLANGE, ZLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLABAD, DLASUM, XERBLA, ZGEQR2, ZGET51, ZGET52,
+      EXTERNAL           DLABAD, DLASUM, XERBLA, MKL_INIT_XERBLA, ZGEQR2, ZGET51, ZGET52,
      $                   ZGGHRD, ZHGEQZ, ZLACPY, ZLARFG, ZLASET, ZLATM4,
      $                   ZTGEVC, ZUNM2R
 *     ..
@@ -595,6 +595,8 @@
      $                   9*.FALSE. /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zchkhb.f b/TESTING/EIG/zchkhb.f
index a4c658ff3..5bfe90038 100644
--- a/TESTING/EIG/zchkhb.f
+++ b/TESTING/EIG/zchkhb.f
@@ -346,7 +346,7 @@
       EXTERNAL           DLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLASUM, XERBLA, ZHBT21, ZHBTRD, ZLACPY, ZLASET,
+      EXTERNAL           DLASUM, XERBLA, MKL_INIT_XERBLA, ZHBT21, ZHBTRD, ZLACPY, ZLASET,
      $                   ZLATMR, ZLATMS
 *     ..
 *     .. Intrinsic Functions ..
@@ -360,6 +360,8 @@
      $                   0, 0 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zchkhb2stg.f b/TESTING/EIG/zchkhb2stg.f
index dbbb84348..e71a59a13 100644
--- a/TESTING/EIG/zchkhb2stg.f
+++ b/TESTING/EIG/zchkhb2stg.f
@@ -372,7 +372,7 @@
       EXTERNAL           DLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLASUM, XERBLA, ZHBT21, ZHBTRD, ZLACPY, ZLASET,
+      EXTERNAL           DLASUM, XERBLA, MKL_INIT_XERBLA, ZHBT21, ZHBTRD, ZLACPY, ZLASET,
      $                   ZLATMR, ZLATMS, ZHETRD_HB2ST, ZSTEQR
 *     ..
 *     .. Intrinsic Functions ..
@@ -386,6 +386,8 @@
      $                   0, 0 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zchkhs.f b/TESTING/EIG/zchkhs.f
index ec89b44f9..1b822245a 100644
--- a/TESTING/EIG/zchkhs.f
+++ b/TESTING/EIG/zchkhs.f
@@ -464,7 +464,7 @@
       EXTERNAL           DLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLABAD, DLAFTS, DLASUM, XERBLA, ZCOPY, ZGEHRD,
+      EXTERNAL           DLABAD, DLAFTS, DLASUM, XERBLA, MKL_INIT_XERBLA, ZCOPY, ZGEHRD,
      $                   ZGEMM, ZGET10, ZGET22, ZHSEIN, ZHSEQR, ZHST01,
      $                   ZLACPY, ZLASET, ZLATME, ZLATMR, ZLATMS, ZTREVC,
      $                   ZUNGHR, ZUNMHR
@@ -481,6 +481,8 @@
       DATA               KCONDS / 3*0, 5*0, 4*1, 6*2, 3*0 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zdrges.f b/TESTING/EIG/zdrges.f
index c09984e09..9b11fef82 100644
--- a/TESTING/EIG/zdrges.f
+++ b/TESTING/EIG/zdrges.f
@@ -437,7 +437,7 @@
       EXTERNAL           ZLCTES, ILAENV, DLAMCH, ZLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, DLABAD, XERBLA, ZGET51, ZGET54, ZGGES,
+      EXTERNAL           ALASVM, DLABAD, XERBLA, MKL_INIT_XERBLA, ZGET51, ZGET54, ZGGES,
      $                   ZLACPY, ZLARFG, ZLASET, ZLATM4, ZUNM2R
 *     ..
 *     .. Intrinsic Functions ..
@@ -475,6 +475,8 @@
      $                   9*.FALSE. /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zdrges3.f b/TESTING/EIG/zdrges3.f
index 2b3be003c..f360e3177 100644
--- a/TESTING/EIG/zdrges3.f
+++ b/TESTING/EIG/zdrges3.f
@@ -438,7 +438,7 @@
       EXTERNAL           ZLCTES, ILAENV, DLAMCH, ZLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, DLABAD, XERBLA, ZGET51, ZGET54, ZGGES3,
+      EXTERNAL           ALASVM, DLABAD, XERBLA, MKL_INIT_XERBLA, ZGET51, ZGET54, ZGGES3,
      $                   ZLACPY, ZLARFG, ZLASET, ZLATM4, ZUNM2R
 *     ..
 *     .. Intrinsic Functions ..
@@ -476,6 +476,8 @@
      $                   9*.FALSE. /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zdrgev.f b/TESTING/EIG/zdrgev.f
index 2588816a5..136310f6a 100644
--- a/TESTING/EIG/zdrgev.f
+++ b/TESTING/EIG/zdrgev.f
@@ -454,7 +454,7 @@
       EXTERNAL           ILAENV, DLAMCH, ZLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, DLABAD, XERBLA, ZGET52, ZGGEV, ZLACPY,
+      EXTERNAL           ALASVM, DLABAD, XERBLA, MKL_INIT_XERBLA, ZGET52, ZGGEV, ZLACPY,
      $                   ZLARFG, ZLASET, ZLATM4, ZUNM2R
 *     ..
 *     .. Intrinsic Functions ..
@@ -486,6 +486,8 @@
      $                   9*.FALSE. /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zdrgev3.f b/TESTING/EIG/zdrgev3.f
index 11e8562d7..237ee5f2d 100644
--- a/TESTING/EIG/zdrgev3.f
+++ b/TESTING/EIG/zdrgev3.f
@@ -454,7 +454,7 @@
       EXTERNAL           ILAENV, DLAMCH, ZLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, DLABAD, XERBLA, ZGET52, ZGGEV3, ZLACPY,
+      EXTERNAL           ALASVM, DLABAD, XERBLA, MKL_INIT_XERBLA, ZGET52, ZGGEV3, ZLACPY,
      $                   ZLARFG, ZLASET, ZLATM4, ZUNM2R
 *     ..
 *     .. Intrinsic Functions ..
@@ -486,6 +486,8 @@
      $                   9*.FALSE. /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zdrgsx.f b/TESTING/EIG/zdrgsx.f
index f5821e520..fd706edcc 100644
--- a/TESTING/EIG/zdrgsx.f
+++ b/TESTING/EIG/zdrgsx.f
@@ -397,7 +397,7 @@
       EXTERNAL           ZLCTSX, ILAENV, DLAMCH, ZLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, DLABAD, XERBLA, ZGESVD, ZGET51, ZGGESX,
+      EXTERNAL           ALASVM, DLABAD, XERBLA, MKL_INIT_XERBLA, ZGESVD, ZGET51, ZGGESX,
      $                   ZLACPY, ZLAKF2, ZLASET, ZLATM5
 *     ..
 *     .. Scalars in Common ..
@@ -417,6 +417,8 @@
       ABS1( X ) = ABS( DBLE( X ) ) + ABS( DIMAG( X ) )
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zdrgvx.f b/TESTING/EIG/zdrgvx.f
index 4a6956451..2153e9c99 100644
--- a/TESTING/EIG/zdrgvx.f
+++ b/TESTING/EIG/zdrgvx.f
@@ -339,12 +339,14 @@
       EXTERNAL           ILAENV, DLAMCH, ZLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, XERBLA, ZGET52, ZGGEVX, ZLACPY, ZLATM6
+      EXTERNAL           ALASVM, XERBLA, MKL_INIT_XERBLA, ZGET52, ZGGEVX, ZLACPY, ZLATM6
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DCMPLX, MAX, SQRT
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zdrvbd.f b/TESTING/EIG/zdrvbd.f
index 105e9dff7..0086d07b5 100644
--- a/TESTING/EIG/zdrvbd.f
+++ b/TESTING/EIG/zdrvbd.f
@@ -458,7 +458,7 @@
       EXTERNAL           DLAMCH, DLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, XERBLA, ZBDT01, ZBDT05, ZGESDD,
+      EXTERNAL           ALASVM, XERBLA, MKL_INIT_XERBLA, ZBDT01, ZBDT05, ZGESDD,
      $                   ZGESVD, ZGESVDQ, ZGESVJ, ZGEJSV, ZGESVDX,
      $                   ZLACPY, ZLASET, ZLATMS, ZUNT01, ZUNT03
 *     ..
@@ -477,6 +477,8 @@
       DATA               CJOBV / 'N', 'V' /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zdrves.f b/TESTING/EIG/zdrves.f
index 7a625e1b7..b90bb2f57 100644
--- a/TESTING/EIG/zdrves.f
+++ b/TESTING/EIG/zdrves.f
@@ -440,7 +440,7 @@
       EXTERNAL           ZSLECT, DLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLABAD, DLASUM, XERBLA, ZGEES, ZHST01, ZLACPY,
+      EXTERNAL           DLABAD, DLASUM, XERBLA, MKL_INIT_XERBLA, ZGEES, ZHST01, ZLACPY,
      $                   ZLASET, ZLATME, ZLATMR, ZLATMS
 *     ..
 *     .. Intrinsic Functions ..
@@ -455,6 +455,8 @@
       DATA               KCONDS / 3*0, 5*0, 4*1, 6*2, 3*0 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       PATH( 1: 1 ) = 'Zomplex precision'
       PATH( 2: 3 ) = 'ES'
diff --git a/TESTING/EIG/zdrvev.f b/TESTING/EIG/zdrvev.f
index e818f7ea6..6a90f68ca 100644
--- a/TESTING/EIG/zdrvev.f
+++ b/TESTING/EIG/zdrvev.f
@@ -445,7 +445,7 @@
       EXTERNAL           DLAMCH, DZNRM2
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLABAD, DLASUM, XERBLA, ZGEEV, ZGET22, ZLACPY,
+      EXTERNAL           DLABAD, DLASUM, XERBLA, MKL_INIT_XERBLA, ZGEEV, ZGET22, ZLACPY,
      $                   ZLASET, ZLATME, ZLATMR, ZLATMS
 *     ..
 *     .. Intrinsic Functions ..
@@ -460,6 +460,8 @@
       DATA               KCONDS / 3*0, 5*0, 4*1, 6*2, 3*0 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       PATH( 1: 1 ) = 'Zomplex precision'
       PATH( 2: 3 ) = 'EV'
diff --git a/TESTING/EIG/zdrvsg.f b/TESTING/EIG/zdrvsg.f
index 8238dceef..d1b62ef22 100644
--- a/TESTING/EIG/zdrvsg.f
+++ b/TESTING/EIG/zdrvsg.f
@@ -421,7 +421,7 @@
       EXTERNAL           LSAME, DLAMCH, DLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLABAD, DLAFTS, DLASUM, XERBLA, ZHBGV, ZHBGVD,
+      EXTERNAL           DLABAD, DLAFTS, DLASUM, XERBLA, MKL_INIT_XERBLA, ZHBGV, ZHBGVD,
      $                   ZHBGVX, ZHEGV, ZHEGVD, ZHEGVX, ZHPGV, ZHPGVD,
      $                   ZHPGVX, ZLACPY, ZLASET, ZLATMR, ZLATMS, ZSGT01
 *     ..
@@ -436,6 +436,8 @@
      $                   0, 0, 6*4 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Check for errors
 *
diff --git a/TESTING/EIG/zdrvsg2stg.f b/TESTING/EIG/zdrvsg2stg.f
index 8929b33c4..4a5306ec5 100644
--- a/TESTING/EIG/zdrvsg2stg.f
+++ b/TESTING/EIG/zdrvsg2stg.f
@@ -429,7 +429,7 @@
       EXTERNAL           LSAME, DLAMCH, DLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLABAD, DLAFTS, DLASUM, XERBLA, ZHBGV, ZHBGVD,
+      EXTERNAL           DLABAD, DLAFTS, DLASUM, XERBLA, MKL_INIT_XERBLA, ZHBGV, ZHBGVD,
      $                   ZHBGVX, ZHEGV, ZHEGVD, ZHEGVX, ZHPGV, ZHPGVD,
      $                   ZHPGVX, ZLACPY, ZLASET, ZLATMR, ZLATMS, ZSGT01,
      $                   ZHEGV_2STAGE
@@ -445,6 +445,8 @@
      $                   0, 0, 6*4 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Check for errors
 *
diff --git a/TESTING/EIG/zdrvst.f b/TESTING/EIG/zdrvst.f
index cf2dc03ed..d1d79a05e 100644
--- a/TESTING/EIG/zdrvst.f
+++ b/TESTING/EIG/zdrvst.f
@@ -394,7 +394,7 @@
       EXTERNAL           DLAMCH, DLARND, DSXT1
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, DLABAD, DLAFTS, XERBLA, ZHBEV, ZHBEVD,
+      EXTERNAL           ALASVM, DLABAD, DLAFTS, XERBLA, MKL_INIT_XERBLA, ZHBEV, ZHBEVD,
      $                   ZHBEVX, ZHEEV, ZHEEVD, ZHEEVR, ZHEEVX, ZHET21,
      $                   ZHET22, ZHPEV, ZHPEVD, ZHPEVX, ZLACPY, ZLASET,
      $                   ZLATMR, ZLATMS
@@ -410,6 +410,8 @@
      $                   0, 0, 4, 4, 4 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Check for errors
 *
diff --git a/TESTING/EIG/zdrvst2stg.f b/TESTING/EIG/zdrvst2stg.f
index de072c9a3..8503f4240 100644
--- a/TESTING/EIG/zdrvst2stg.f
+++ b/TESTING/EIG/zdrvst2stg.f
@@ -394,7 +394,7 @@
       EXTERNAL           DLAMCH, DLARND, DSXT1
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ALASVM, DLABAD, DLAFTS, XERBLA, ZHBEV, ZHBEVD,
+      EXTERNAL           ALASVM, DLABAD, DLAFTS, XERBLA, MKL_INIT_XERBLA, ZHBEV, ZHBEVD,
      $                   ZHBEVX, ZHEEV, ZHEEVD, ZHEEVR, ZHEEVX, ZHET21,
      $                   ZHET22, ZHPEV, ZHPEVD, ZHPEVX, ZLACPY, ZLASET,
      $                   ZHEEVD_2STAGE, ZHEEVR_2STAGE, ZHEEVX_2STAGE,
@@ -412,6 +412,8 @@
      $                   0, 0, 4, 4, 4 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Check for errors
 *
diff --git a/TESTING/EIG/zdrvsx.f b/TESTING/EIG/zdrvsx.f
index ba3ffab6c..256330661 100644
--- a/TESTING/EIG/zdrvsx.f
+++ b/TESTING/EIG/zdrvsx.f
@@ -495,7 +495,7 @@
       EXTERNAL           DLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLABAD, DLASUM, XERBLA, ZGET24, ZLASET, ZLATME,
+      EXTERNAL           DLABAD, DLASUM, XERBLA, MKL_INIT_XERBLA, ZGET24, ZLASET, ZLATME,
      $                   ZLATMR, ZLATMS
 *     ..
 *     .. Intrinsic Functions ..
@@ -510,6 +510,8 @@
       DATA               KCONDS / 3*0, 5*0, 4*1, 6*2, 3*0 /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       PATH( 1: 1 ) = 'Zomplex precision'
       PATH( 2: 3 ) = 'SX'
diff --git a/TESTING/EIG/zdrvvx.f b/TESTING/EIG/zdrvvx.f
index 9492e0623..e9f59e008 100644
--- a/TESTING/EIG/zdrvvx.f
+++ b/TESTING/EIG/zdrvvx.f
@@ -551,7 +551,7 @@
       EXTERNAL           DLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLABAD, DLASUM, XERBLA, ZGET23, ZLASET, ZLATME,
+      EXTERNAL           DLABAD, DLASUM, XERBLA, MKL_INIT_XERBLA, ZGET23, ZLASET, ZLATME,
      $                   ZLATMR, ZLATMS
 *     ..
 *     .. Intrinsic Functions ..
@@ -567,6 +567,8 @@
       DATA               BAL / 'N', 'P', 'S', 'B' /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       PATH( 1: 1 ) = 'Zomplex precision'
       PATH( 2: 3 ) = 'VX'
diff --git a/TESTING/EIG/zget23.f b/TESTING/EIG/zget23.f
index 39fe15299..a0ce79d3a 100644
--- a/TESTING/EIG/zget23.f
+++ b/TESTING/EIG/zget23.f
@@ -420,7 +420,7 @@
       EXTERNAL           LSAME, DLAMCH, DZNRM2
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGEEVX, ZGET22, ZLACPY
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGEEVX, ZGET22, ZLACPY
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, DIMAG, MAX, MIN
@@ -429,6 +429,8 @@
       DATA               SENS / 'N', 'V' /
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zget24.f b/TESTING/EIG/zget24.f
index 1641a018c..fd0b9bdbd 100644
--- a/TESTING/EIG/zget24.f
+++ b/TESTING/EIG/zget24.f
@@ -384,7 +384,7 @@
       EXTERNAL           ZSLECT, DLAMCH, ZLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZCOPY, ZGEESX, ZGEMM, ZLACPY, ZUNT01
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZCOPY, ZGEESX, ZGEMM, ZLACPY, ZUNT01
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, DIMAG, MAX, MIN
@@ -400,6 +400,8 @@
       COMMON             / SSLCT / SELOPT, SELDIM, SELVAL, SELWR, SELWI
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check for errors
 *
diff --git a/TESTING/EIG/zlarhs.f b/TESTING/EIG/zlarhs.f
index f2fe47e9b..4d13db5f9 100644
--- a/TESTING/EIG/zlarhs.f
+++ b/TESTING/EIG/zlarhs.f
@@ -242,7 +242,7 @@
       EXTERNAL           LSAME, LSAMEN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGBMV, ZGEMM, ZHBMV, ZHEMM, ZHPMV,
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGBMV, ZGEMM, ZHBMV, ZHEMM, ZHPMV,
      $                   ZLACPY, ZLARNV, ZSBMV, ZSPMV, ZSYMM, ZTBMV,
      $                   ZTPMV, ZTRMM
 *     ..
@@ -250,6 +250,8 @@
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/EIG/zsbmv.f b/TESTING/EIG/zsbmv.f
index 643882550..b88228f85 100644
--- a/TESTING/EIG/zsbmv.f
+++ b/TESTING/EIG/zsbmv.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, MIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/EIG/zunt03.f b/TESTING/EIG/zunt03.f
index 4560410a3..cf1883c0a 100644
--- a/TESTING/EIG/zunt03.f
+++ b/TESTING/EIG/zunt03.f
@@ -199,9 +199,11 @@
       INTRINSIC          ABS, DBLE, DCMPLX, MAX, MIN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZUNT01
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZUNT01
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Check inputs
 *
diff --git a/TESTING/LIN/cgelqs.f b/TESTING/LIN/cgelqs.f
index ee08c9b62..de1ca3a5c 100644
--- a/TESTING/LIN/cgelqs.f
+++ b/TESTING/LIN/cgelqs.f
@@ -142,12 +142,14 @@
      $                   CONE = ( 1.0E+0, 0.0E+0 ) )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CLASET, CTRSM, CUNMLQ, XERBLA
+      EXTERNAL           CLASET, CTRSM, CUNMLQ, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/cgeqls.f b/TESTING/LIN/cgeqls.f
index b1d36107c..7fcf01e70 100644
--- a/TESTING/LIN/cgeqls.f
+++ b/TESTING/LIN/cgeqls.f
@@ -142,12 +142,14 @@
       PARAMETER          ( ONE = ( 1.0E+0, 0.0E+0 ) )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CTRSM, CUNMQL, XERBLA
+      EXTERNAL           CTRSM, CUNMQL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments.
 *
diff --git a/TESTING/LIN/cgeqrs.f b/TESTING/LIN/cgeqrs.f
index 76982924c..eba74dd24 100644
--- a/TESTING/LIN/cgeqrs.f
+++ b/TESTING/LIN/cgeqrs.f
@@ -141,12 +141,14 @@
       PARAMETER          ( ONE = ( 1.0E+0, 0.0E+0 ) )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CTRSM, CUNMQR, XERBLA
+      EXTERNAL           CTRSM, CUNMQR, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments.
 *
diff --git a/TESTING/LIN/cgerqs.f b/TESTING/LIN/cgerqs.f
index 5603ed478..302111ffd 100644
--- a/TESTING/LIN/cgerqs.f
+++ b/TESTING/LIN/cgerqs.f
@@ -143,12 +143,14 @@
      $                   CONE = ( 1.0E+0, 0.0E+0 ) )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CLASET, CTRSM, CUNMRQ, XERBLA
+      EXTERNAL           CLASET, CTRSM, CUNMRQ, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/clavhe.f b/TESTING/LIN/clavhe.f
index 0a5016778..971d4fe7e 100644
--- a/TESTING/LIN/clavhe.f
+++ b/TESTING/LIN/clavhe.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGEMV, CGERU, CLACGV, CSCAL, CSWAP, XERBLA
+      EXTERNAL           CGEMV, CGERU, CLACGV, CSCAL, CSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, CONJG, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/clavhe_rook.f b/TESTING/LIN/clavhe_rook.f
index 6fca41618..3b0cf2458 100644
--- a/TESTING/LIN/clavhe_rook.f
+++ b/TESTING/LIN/clavhe_rook.f
@@ -186,12 +186,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGEMV, CGERU, CLACGV, CSCAL, CSWAP, XERBLA
+      EXTERNAL           CGEMV, CGERU, CLACGV, CSCAL, CSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, CONJG, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/clavhp.f b/TESTING/LIN/clavhp.f
index 1d0176cbc..a297457de 100644
--- a/TESTING/LIN/clavhp.f
+++ b/TESTING/LIN/clavhp.f
@@ -161,12 +161,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGEMV, CGERU, CLACGV, CSCAL, CSWAP, XERBLA
+      EXTERNAL           CGEMV, CGERU, CLACGV, CSCAL, CSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, CONJG, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/clavsp.f b/TESTING/LIN/clavsp.f
index 311f0b4e1..0003a29f0 100644
--- a/TESTING/LIN/clavsp.f
+++ b/TESTING/LIN/clavsp.f
@@ -161,12 +161,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGEMV, CGERU, CSCAL, CSWAP, XERBLA
+      EXTERNAL           CGEMV, CGERU, CSCAL, CSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/clavsy.f b/TESTING/LIN/clavsy.f
index c4d885cc0..2426232bb 100644
--- a/TESTING/LIN/clavsy.f
+++ b/TESTING/LIN/clavsy.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGEMV, CGERU, CSCAL, CSWAP, XERBLA
+      EXTERNAL           CGEMV, CGERU, CSCAL, CSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/clavsy_rook.f b/TESTING/LIN/clavsy_rook.f
index 73f6e9acc..791d86f68 100644
--- a/TESTING/LIN/clavsy_rook.f
+++ b/TESTING/LIN/clavsy_rook.f
@@ -185,12 +185,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGEMV, CGERU, CSCAL, CSWAP, XERBLA
+      EXTERNAL           CGEMV, CGERU, CSCAL, CSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/cqpt01.f b/TESTING/LIN/cqpt01.f
index aa9ff4d91..99e336b09 100644
--- a/TESTING/LIN/cqpt01.f
+++ b/TESTING/LIN/cqpt01.f
@@ -152,12 +152,14 @@
       EXTERNAL           CLANGE, SLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CAXPY, CCOPY, CUNMQR, XERBLA
+      EXTERNAL           CAXPY, CCOPY, CUNMQR, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          CMPLX, MAX, MIN, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       CQPT01 = ZERO
 *
diff --git a/TESTING/LIN/cqrt11.f b/TESTING/LIN/cqrt11.f
index 7445059a4..4c9ee553f 100644
--- a/TESTING/LIN/cqrt11.f
+++ b/TESTING/LIN/cqrt11.f
@@ -124,7 +124,7 @@
       EXTERNAL           CLANGE, SLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CLASET, CUNM2R, XERBLA
+      EXTERNAL           CLASET, CUNM2R, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          CMPLX, REAL
@@ -133,6 +133,8 @@
       REAL               RDUMMY( 1 )
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       CQRT11 = ZERO
 *
diff --git a/TESTING/LIN/cqrt14.f b/TESTING/LIN/cqrt14.f
index b2a31e99d..f1cf351b5 100644
--- a/TESTING/LIN/cqrt14.f
+++ b/TESTING/LIN/cqrt14.f
@@ -149,12 +149,14 @@
       EXTERNAL           LSAME, CLANGE, SLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGELQ2, CGEQR2, CLACPY, CLASCL, XERBLA
+      EXTERNAL           CGELQ2, CGEQR2, CLACPY, CLASCL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, CONJG, MAX, MIN, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       CQRT14 = ZERO
       IF( LSAME( TRANS, 'N' ) ) THEN
diff --git a/TESTING/LIN/cqrt17.f b/TESTING/LIN/cqrt17.f
index 1fe28d6fe..b52a4778d 100644
--- a/TESTING/LIN/cqrt17.f
+++ b/TESTING/LIN/cqrt17.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME, CLANGE, SLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGEMM, CLACPY, CLASCL, XERBLA
+      EXTERNAL           CGEMM, CLACPY, CLASCL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          CMPLX, MAX, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       CQRT17 = ZERO
 *
diff --git a/TESTING/LIN/crzt01.f b/TESTING/LIN/crzt01.f
index 73141982a..32dc05222 100644
--- a/TESTING/LIN/crzt01.f
+++ b/TESTING/LIN/crzt01.f
@@ -129,12 +129,14 @@
       EXTERNAL           CLANGE, SLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CAXPY, CLASET, CUNMRZ, XERBLA
+      EXTERNAL           CAXPY, CLASET, CUNMRZ, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          CMPLX, MAX, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       CRZT01 = ZERO
 *
diff --git a/TESTING/LIN/crzt02.f b/TESTING/LIN/crzt02.f
index 53a69457c..d8d27dace 100644
--- a/TESTING/LIN/crzt02.f
+++ b/TESTING/LIN/crzt02.f
@@ -120,12 +120,14 @@
       EXTERNAL           CLANGE, SLAMCH
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CLASET, CUNMRZ, XERBLA
+      EXTERNAL           CLASET, CUNMRZ, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          CMPLX, MAX, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       CRZT02 = ZERO
 *
diff --git a/TESTING/LIN/csbmv.f b/TESTING/LIN/csbmv.f
index 776de784a..c2d428350 100644
--- a/TESTING/LIN/csbmv.f
+++ b/TESTING/LIN/csbmv.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, MIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/dgelqs.f b/TESTING/LIN/dgelqs.f
index 46683ef12..f397a4f05 100644
--- a/TESTING/LIN/dgelqs.f
+++ b/TESTING/LIN/dgelqs.f
@@ -141,12 +141,14 @@
       PARAMETER          ( ZERO = 0.0D+0, ONE = 1.0D+0 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLASET, DORMLQ, DTRSM, XERBLA
+      EXTERNAL           DLASET, DORMLQ, DTRSM, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/dgeqls.f b/TESTING/LIN/dgeqls.f
index 1ca418f1b..265fd2911 100644
--- a/TESTING/LIN/dgeqls.f
+++ b/TESTING/LIN/dgeqls.f
@@ -142,12 +142,14 @@
       PARAMETER          ( ONE = 1.0D+0 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DORMQL, DTRSM, XERBLA
+      EXTERNAL           DORMQL, DTRSM, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments.
 *
diff --git a/TESTING/LIN/dgeqrs.f b/TESTING/LIN/dgeqrs.f
index 83fddf4e1..786741f3c 100644
--- a/TESTING/LIN/dgeqrs.f
+++ b/TESTING/LIN/dgeqrs.f
@@ -141,12 +141,14 @@
       PARAMETER          ( ONE = 1.0D+0 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DORMQR, DTRSM, XERBLA
+      EXTERNAL           DORMQR, DTRSM, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments.
 *
diff --git a/TESTING/LIN/dgerqs.f b/TESTING/LIN/dgerqs.f
index d3ea086fa..690964f06 100644
--- a/TESTING/LIN/dgerqs.f
+++ b/TESTING/LIN/dgerqs.f
@@ -142,12 +142,14 @@
       PARAMETER          ( ZERO = 0.0D+0, ONE = 1.0D+0 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLASET, DORMRQ, DTRSM, XERBLA
+      EXTERNAL           DLASET, DORMRQ, DTRSM, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/dlavsp.f b/TESTING/LIN/dlavsp.f
index 758c415fb..e0d1caf3c 100644
--- a/TESTING/LIN/dlavsp.f
+++ b/TESTING/LIN/dlavsp.f
@@ -160,12 +160,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DGEMV, DGER, DSCAL, DSWAP, XERBLA
+      EXTERNAL           DGEMV, DGER, DSCAL, DSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/dlavsy.f b/TESTING/LIN/dlavsy.f
index b512ff0ff..22c5de42c 100644
--- a/TESTING/LIN/dlavsy.f
+++ b/TESTING/LIN/dlavsy.f
@@ -185,12 +185,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DGEMV, DGER, DSCAL, DSWAP, XERBLA
+      EXTERNAL           DGEMV, DGER, DSCAL, DSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/dlavsy_rook.f b/TESTING/LIN/dlavsy_rook.f
index 7b0c2c152..1cc611e82 100644
--- a/TESTING/LIN/dlavsy_rook.f
+++ b/TESTING/LIN/dlavsy_rook.f
@@ -187,12 +187,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DGEMV, DGER, DSCAL, DSWAP, XERBLA
+      EXTERNAL           DGEMV, DGER, DSCAL, DSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/dqpt01.f b/TESTING/LIN/dqpt01.f
index b72eaaa77..ade0661d9 100644
--- a/TESTING/LIN/dqpt01.f
+++ b/TESTING/LIN/dqpt01.f
@@ -152,12 +152,14 @@
       EXTERNAL           DLAMCH, DLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DAXPY, DCOPY, DORMQR, XERBLA
+      EXTERNAL           DAXPY, DCOPY, DORMQR, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DBLE, MAX, MIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       DQPT01 = ZERO
 *
diff --git a/TESTING/LIN/dqrt11.f b/TESTING/LIN/dqrt11.f
index fe29ca589..fbf2f1b90 100644
--- a/TESTING/LIN/dqrt11.f
+++ b/TESTING/LIN/dqrt11.f
@@ -124,7 +124,7 @@
       EXTERNAL           DLAMCH, DLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLASET, DORM2R, XERBLA
+      EXTERNAL           DLASET, DORM2R, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DBLE
@@ -133,6 +133,8 @@
       DOUBLE PRECISION   RDUMMY( 1 )
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       DQRT11 = ZERO
 *
diff --git a/TESTING/LIN/dqrt14.f b/TESTING/LIN/dqrt14.f
index 9301e251f..aa6d2bb1e 100644
--- a/TESTING/LIN/dqrt14.f
+++ b/TESTING/LIN/dqrt14.f
@@ -149,12 +149,14 @@
       EXTERNAL           LSAME, DLAMCH, DLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DGELQ2, DGEQR2, DLACPY, DLASCL, XERBLA
+      EXTERNAL           DGELQ2, DGEQR2, DLACPY, DLASCL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, MAX, MIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       DQRT14 = ZERO
       IF( LSAME( TRANS, 'N' ) ) THEN
diff --git a/TESTING/LIN/dqrt17.f b/TESTING/LIN/dqrt17.f
index 78580dca1..f62e67952 100644
--- a/TESTING/LIN/dqrt17.f
+++ b/TESTING/LIN/dqrt17.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME, DLAMCH, DLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DGEMM, DLACPY, DLASCL, XERBLA
+      EXTERNAL           DGEMM, DLACPY, DLASCL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DBLE, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       DQRT17 = ZERO
 *
diff --git a/TESTING/LIN/drzt01.f b/TESTING/LIN/drzt01.f
index 2108cd1c9..5b001e132 100644
--- a/TESTING/LIN/drzt01.f
+++ b/TESTING/LIN/drzt01.f
@@ -129,12 +129,14 @@
       EXTERNAL           DLAMCH, DLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DAXPY, DLASET, DORMRZ, XERBLA
+      EXTERNAL           DAXPY, DLASET, DORMRZ, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DBLE, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       DRZT01 = ZERO
 *
diff --git a/TESTING/LIN/drzt02.f b/TESTING/LIN/drzt02.f
index de3f26621..76b86ab4d 100644
--- a/TESTING/LIN/drzt02.f
+++ b/TESTING/LIN/drzt02.f
@@ -120,12 +120,14 @@
       EXTERNAL           DLAMCH, DLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLASET, DORMRZ, XERBLA
+      EXTERNAL           DLASET, DORMRZ, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DBLE, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       DRZT02 = ZERO
 *
diff --git a/TESTING/LIN/dtplqt.f b/TESTING/LIN/dtplqt.f
index 9fcecc984..1be638687 100644
--- a/TESTING/LIN/dtplqt.f
+++ b/TESTING/LIN/dtplqt.f
@@ -208,9 +208,11 @@
       INTEGER    I, IB, LB, NB, IINFO
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL   DTPLQT2, DTPRFB, XERBLA
+      EXTERNAL   DTPLQT2, DTPRFB, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/LIN/sgelqs.f b/TESTING/LIN/sgelqs.f
index cdffebdc8..f652af627 100644
--- a/TESTING/LIN/sgelqs.f
+++ b/TESTING/LIN/sgelqs.f
@@ -141,12 +141,14 @@
       PARAMETER          ( ZERO = 0.0E+0, ONE = 1.0E+0 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SLASET, SORMLQ, STRSM, XERBLA
+      EXTERNAL           SLASET, SORMLQ, STRSM, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/sgeqls.f b/TESTING/LIN/sgeqls.f
index 19ae7c592..722b68727 100644
--- a/TESTING/LIN/sgeqls.f
+++ b/TESTING/LIN/sgeqls.f
@@ -142,12 +142,14 @@
       PARAMETER          ( ONE = 1.0E+0 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SORMQL, STRSM, XERBLA
+      EXTERNAL           SORMQL, STRSM, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments.
 *
diff --git a/TESTING/LIN/sgeqrs.f b/TESTING/LIN/sgeqrs.f
index 9ec64073d..04d54b38e 100644
--- a/TESTING/LIN/sgeqrs.f
+++ b/TESTING/LIN/sgeqrs.f
@@ -141,12 +141,14 @@
       PARAMETER          ( ONE = 1.0E+0 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SORMQR, STRSM, XERBLA
+      EXTERNAL           SORMQR, STRSM, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments.
 *
diff --git a/TESTING/LIN/sgerqs.f b/TESTING/LIN/sgerqs.f
index 5b04c5ae2..88de76c97 100644
--- a/TESTING/LIN/sgerqs.f
+++ b/TESTING/LIN/sgerqs.f
@@ -142,12 +142,14 @@
       PARAMETER          ( ZERO = 0.0E+0, ONE = 1.0E+0 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SLASET, SORMRQ, STRSM, XERBLA
+      EXTERNAL           SLASET, SORMRQ, STRSM, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/slavsp.f b/TESTING/LIN/slavsp.f
index 208a86c9f..7bf81faef 100644
--- a/TESTING/LIN/slavsp.f
+++ b/TESTING/LIN/slavsp.f
@@ -160,12 +160,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SGEMV, SGER, SSCAL, SSWAP, XERBLA
+      EXTERNAL           SGEMV, SGER, SSCAL, SSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/slavsy.f b/TESTING/LIN/slavsy.f
index 3c575e2d7..0518c96f4 100644
--- a/TESTING/LIN/slavsy.f
+++ b/TESTING/LIN/slavsy.f
@@ -185,12 +185,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SGEMV, SGER, SSCAL, SSWAP, XERBLA
+      EXTERNAL           SGEMV, SGER, SSCAL, SSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/slavsy_rook.f b/TESTING/LIN/slavsy_rook.f
index 55477bd96..e6665d1d1 100644
--- a/TESTING/LIN/slavsy_rook.f
+++ b/TESTING/LIN/slavsy_rook.f
@@ -187,12 +187,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SGEMV, SGER, SSCAL, SSWAP, XERBLA
+      EXTERNAL           SGEMV, SGER, SSCAL, SSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/sqpt01.f b/TESTING/LIN/sqpt01.f
index e33d841fc..a0ea9b5a1 100644
--- a/TESTING/LIN/sqpt01.f
+++ b/TESTING/LIN/sqpt01.f
@@ -152,12 +152,14 @@
       EXTERNAL           SLAMCH, SLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SAXPY, SCOPY, SORMQR, XERBLA
+      EXTERNAL           SAXPY, SCOPY, SORMQR, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, MIN, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       SQPT01 = ZERO
 *
diff --git a/TESTING/LIN/sqrt11.f b/TESTING/LIN/sqrt11.f
index eaffacbf0..a4305d8a0 100644
--- a/TESTING/LIN/sqrt11.f
+++ b/TESTING/LIN/sqrt11.f
@@ -124,7 +124,7 @@
       EXTERNAL           SLAMCH, SLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SLASET, SORM2R, XERBLA
+      EXTERNAL           SLASET, SORM2R, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          REAL
@@ -133,6 +133,8 @@
       REAL               RDUMMY( 1 )
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       SQRT11 = ZERO
 *
diff --git a/TESTING/LIN/sqrt14.f b/TESTING/LIN/sqrt14.f
index 6e8ac2f3e..db4f2e976 100644
--- a/TESTING/LIN/sqrt14.f
+++ b/TESTING/LIN/sqrt14.f
@@ -149,12 +149,14 @@
       EXTERNAL           LSAME, SLAMCH, SLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SGELQ2, SGEQR2, SLACPY, SLASCL, XERBLA
+      EXTERNAL           SGELQ2, SGEQR2, SLACPY, SLASCL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX, MIN, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       SQRT14 = ZERO
       IF( LSAME( TRANS, 'N' ) ) THEN
diff --git a/TESTING/LIN/sqrt17.f b/TESTING/LIN/sqrt17.f
index c7b543cb6..342a68716 100644
--- a/TESTING/LIN/sqrt17.f
+++ b/TESTING/LIN/sqrt17.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME, SLAMCH, SLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SGEMM, SLACPY, SLASCL, XERBLA
+      EXTERNAL           SGEMM, SLACPY, SLASCL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       SQRT17 = ZERO
 *
diff --git a/TESTING/LIN/srzt01.f b/TESTING/LIN/srzt01.f
index 549bca2e0..1ead43330 100644
--- a/TESTING/LIN/srzt01.f
+++ b/TESTING/LIN/srzt01.f
@@ -129,12 +129,14 @@
       EXTERNAL           SLAMCH, SLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SAXPY, SLASET, SORMRZ, XERBLA
+      EXTERNAL           SAXPY, SLASET, SORMRZ, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       SRZT01 = ZERO
 *
diff --git a/TESTING/LIN/srzt02.f b/TESTING/LIN/srzt02.f
index 12f4018ff..347c096ee 100644
--- a/TESTING/LIN/srzt02.f
+++ b/TESTING/LIN/srzt02.f
@@ -120,12 +120,14 @@
       EXTERNAL           SLAMCH, SLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SLASET, SORMRZ, XERBLA
+      EXTERNAL           SLASET, SORMRZ, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       SRZT02 = ZERO
 *
diff --git a/TESTING/LIN/stplqt.f b/TESTING/LIN/stplqt.f
index 30f827998..6c8f874b4 100644
--- a/TESTING/LIN/stplqt.f
+++ b/TESTING/LIN/stplqt.f
@@ -191,9 +191,11 @@
       INTEGER    I, IB, LB, NB, IINFO
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL   STPLQT2, STPRFB, XERBLA
+      EXTERNAL   STPLQT2, STPRFB, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/LIN/zgelqs.f b/TESTING/LIN/zgelqs.f
index 7545d015e..2cf74cfc4 100644
--- a/TESTING/LIN/zgelqs.f
+++ b/TESTING/LIN/zgelqs.f
@@ -142,12 +142,14 @@
      $                   CONE = ( 1.0D+0, 0.0D+0 ) )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZLASET, ZTRSM, ZUNMLQ
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZLASET, ZTRSM, ZUNMLQ
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/zgeqls.f b/TESTING/LIN/zgeqls.f
index b668fac5d..b95ae8260 100644
--- a/TESTING/LIN/zgeqls.f
+++ b/TESTING/LIN/zgeqls.f
@@ -142,12 +142,14 @@
       PARAMETER          ( ONE = ( 1.0D+0, 0.0D+0 ) )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZTRSM, ZUNMQL
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZTRSM, ZUNMQL
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments.
 *
diff --git a/TESTING/LIN/zgeqrs.f b/TESTING/LIN/zgeqrs.f
index f0d020817..9ae808c82 100644
--- a/TESTING/LIN/zgeqrs.f
+++ b/TESTING/LIN/zgeqrs.f
@@ -141,12 +141,14 @@
       PARAMETER          ( ONE = ( 1.0D+0, 0.0D+0 ) )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZTRSM, ZUNMQR
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZTRSM, ZUNMQR
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments.
 *
diff --git a/TESTING/LIN/zgerqs.f b/TESTING/LIN/zgerqs.f
index d94a7a3dc..a38f2dc99 100644
--- a/TESTING/LIN/zgerqs.f
+++ b/TESTING/LIN/zgerqs.f
@@ -143,12 +143,14 @@
      $                   CONE = ( 1.0D+0, 0.0D+0 ) )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZLASET, ZTRSM, ZUNMRQ
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZLASET, ZTRSM, ZUNMRQ
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/zlarhs.f b/TESTING/LIN/zlarhs.f
index 9d1452840..9096add33 100644
--- a/TESTING/LIN/zlarhs.f
+++ b/TESTING/LIN/zlarhs.f
@@ -242,7 +242,7 @@
       EXTERNAL           LSAME, LSAMEN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGBMV, ZGEMM, ZHBMV, ZHEMM, ZHPMV,
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGBMV, ZGEMM, ZHBMV, ZHEMM, ZHPMV,
      $                   ZLACPY, ZLARNV, ZSBMV, ZSPMV, ZSYMM, ZTBMV,
      $                   ZTPMV, ZTRMM
 *     ..
@@ -250,6 +250,8 @@
       INTRINSIC          MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/zlavhe.f b/TESTING/LIN/zlavhe.f
index bf58d7d74..a5472b306 100644
--- a/TESTING/LIN/zlavhe.f
+++ b/TESTING/LIN/zlavhe.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGEMV, ZGERU, ZLACGV, ZSCAL, ZSWAP
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGEMV, ZGERU, ZLACGV, ZSCAL, ZSWAP
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DCONJG, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/zlavhe_rook.f b/TESTING/LIN/zlavhe_rook.f
index b8e708f50..a60ba5ad7 100644
--- a/TESTING/LIN/zlavhe_rook.f
+++ b/TESTING/LIN/zlavhe_rook.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           ZGEMV, ZGERU, ZLACGV, ZSCAL, ZSWAP, XERBLA
+      EXTERNAL           ZGEMV, ZGERU, ZLACGV, ZSCAL, ZSWAP, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DCONJG, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/zlavhp.f b/TESTING/LIN/zlavhp.f
index 08c2b4759..3085ca3bc 100644
--- a/TESTING/LIN/zlavhp.f
+++ b/TESTING/LIN/zlavhp.f
@@ -161,12 +161,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGEMV, ZGERU, ZLACGV, ZSCAL, ZSWAP
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGEMV, ZGERU, ZLACGV, ZSCAL, ZSWAP
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DCONJG, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/zlavsp.f b/TESTING/LIN/zlavsp.f
index a248d01d9..134d3ca95 100644
--- a/TESTING/LIN/zlavsp.f
+++ b/TESTING/LIN/zlavsp.f
@@ -161,12 +161,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGEMV, ZGERU, ZSCAL, ZSWAP
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGEMV, ZGERU, ZSCAL, ZSWAP
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/zlavsy.f b/TESTING/LIN/zlavsy.f
index 2f8ecd88f..9847f80d5 100644
--- a/TESTING/LIN/zlavsy.f
+++ b/TESTING/LIN/zlavsy.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGEMV, ZGERU, ZSCAL, ZSWAP
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGEMV, ZGERU, ZSCAL, ZSWAP
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/zlavsy_rook.f b/TESTING/LIN/zlavsy_rook.f
index ed62f8f17..2db782992 100644
--- a/TESTING/LIN/zlavsy_rook.f
+++ b/TESTING/LIN/zlavsy_rook.f
@@ -185,12 +185,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGEMV, ZGERU, ZSCAL, ZSWAP
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGEMV, ZGERU, ZSCAL, ZSWAP
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/LIN/zqpt01.f b/TESTING/LIN/zqpt01.f
index eea178a2e..453943319 100644
--- a/TESTING/LIN/zqpt01.f
+++ b/TESTING/LIN/zqpt01.f
@@ -152,12 +152,14 @@
       EXTERNAL           DLAMCH, ZLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZAXPY, ZCOPY, ZUNMQR
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZAXPY, ZCOPY, ZUNMQR
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DBLE, DCMPLX, MAX, MIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       ZQPT01 = ZERO
 *
diff --git a/TESTING/LIN/zqrt11.f b/TESTING/LIN/zqrt11.f
index c633df2f6..c77375bee 100644
--- a/TESTING/LIN/zqrt11.f
+++ b/TESTING/LIN/zqrt11.f
@@ -124,7 +124,7 @@
       EXTERNAL           DLAMCH, ZLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZLASET, ZUNM2R
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZLASET, ZUNM2R
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DBLE, DCMPLX
@@ -133,6 +133,8 @@
       DOUBLE PRECISION   RDUMMY( 1 )
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       ZQRT11 = ZERO
 *
diff --git a/TESTING/LIN/zqrt12.f b/TESTING/LIN/zqrt12.f
index 5428adbf3..e8403db84 100644
--- a/TESTING/LIN/zqrt12.f
+++ b/TESTING/LIN/zqrt12.f
@@ -128,13 +128,15 @@
       EXTERNAL           DASUM, DLAMCH, DNRM2, ZLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DAXPY, DBDSQR, DLABAD, DLASCL, XERBLA, ZGEBD2,
+      EXTERNAL           DAXPY, DBDSQR, DLABAD, DLASCL, XERBLA, MKL_INIT_XERBLA, ZGEBD2,
      $                   ZLASCL, ZLASET
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DBLE, DCMPLX, MAX, MIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       ZQRT12 = ZERO
 *
diff --git a/TESTING/LIN/zqrt14.f b/TESTING/LIN/zqrt14.f
index 369efc07d..bcd7f03b4 100644
--- a/TESTING/LIN/zqrt14.f
+++ b/TESTING/LIN/zqrt14.f
@@ -149,12 +149,14 @@
       EXTERNAL           LSAME, DLAMCH, ZLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGELQ2, ZGEQR2, ZLACPY, ZLASCL
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGELQ2, ZGEQR2, ZLACPY, ZLASCL
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, DCONJG, MAX, MIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       ZQRT14 = ZERO
       IF( LSAME( TRANS, 'N' ) ) THEN
diff --git a/TESTING/LIN/zqrt15.f b/TESTING/LIN/zqrt15.f
index 0d17313e8..139847108 100644
--- a/TESTING/LIN/zqrt15.f
+++ b/TESTING/LIN/zqrt15.f
@@ -186,13 +186,15 @@
       EXTERNAL           DASUM, DLAMCH, DLARND, DZNRM2, ZLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLABAD, DLAORD, DLASCL, XERBLA, ZDSCAL, ZGEMM,
+      EXTERNAL           DLABAD, DLAORD, DLASCL, XERBLA, MKL_INIT_XERBLA, ZDSCAL, ZGEMM,
      $                   ZLARF, ZLARNV, ZLAROR, ZLASCL, ZLASET
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DCMPLX, MAX, MIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       MN = MIN( M, N )
       IF( LWORK.LT.MAX( M+MN, MN*NRHS, 2*N+M ) ) THEN
diff --git a/TESTING/LIN/zqrt17.f b/TESTING/LIN/zqrt17.f
index ff694c7a1..c58fb5d02 100644
--- a/TESTING/LIN/zqrt17.f
+++ b/TESTING/LIN/zqrt17.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME, DLAMCH, ZLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGEMM, ZLACPY, ZLASCL
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGEMM, ZLACPY, ZLASCL
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DBLE, DCMPLX, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       ZQRT17 = ZERO
 *
diff --git a/TESTING/LIN/zrzt01.f b/TESTING/LIN/zrzt01.f
index 876384069..6e7ca9470 100644
--- a/TESTING/LIN/zrzt01.f
+++ b/TESTING/LIN/zrzt01.f
@@ -129,12 +129,14 @@
       EXTERNAL           DLAMCH, ZLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZAXPY, ZLASET, ZUNMRZ
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZAXPY, ZLASET, ZUNMRZ
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DBLE, DCMPLX, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       ZRZT01 = ZERO
 *
diff --git a/TESTING/LIN/zrzt02.f b/TESTING/LIN/zrzt02.f
index 373ced982..cbf65db2f 100644
--- a/TESTING/LIN/zrzt02.f
+++ b/TESTING/LIN/zrzt02.f
@@ -120,12 +120,14 @@
       EXTERNAL           DLAMCH, ZLANGE
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZLASET, ZUNMRZ
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZLASET, ZUNMRZ
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DBLE, DCMPLX, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       ZRZT02 = ZERO
 *
diff --git a/TESTING/LIN/zsbmv.f b/TESTING/LIN/zsbmv.f
index 005e288e0..96c01301a 100644
--- a/TESTING/LIN/zsbmv.f
+++ b/TESTING/LIN/zsbmv.f
@@ -183,12 +183,14 @@
       EXTERNAL           LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, MIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input parameters.
 *
diff --git a/TESTING/MATGEN/clagge.f b/TESTING/MATGEN/clagge.f
index f0d86fb50..70c853412 100644
--- a/TESTING/MATGEN/clagge.f
+++ b/TESTING/MATGEN/clagge.f
@@ -141,7 +141,7 @@
       COMPLEX            TAU, WA, WB
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGEMV, CGERC, CLACGV, CLARNV, CSCAL, XERBLA
+      EXTERNAL           CGEMV, CGERC, CLACGV, CLARNV, CSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX, MIN, REAL
@@ -151,6 +151,8 @@
       EXTERNAL           SCNRM2
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/clahilb.f b/TESTING/MATGEN/clahilb.f
index f4481fc78..da482d267 100644
--- a/TESTING/MATGEN/clahilb.f
+++ b/TESTING/MATGEN/clahilb.f
@@ -175,7 +175,7 @@
      $     (-.5,.5),(.5,.5),(.5,-.5)/
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL XERBLA
+      EXTERNAL XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. External Functions
       EXTERNAL CLASET, LSAMEN
@@ -183,6 +183,8 @@
       LOGICAL LSAMEN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
       C2 = PATH( 2: 3 )
 *
 *     Test the input arguments
diff --git a/TESTING/MATGEN/clarge.f b/TESTING/MATGEN/clarge.f
index 4c1d47137..982519b49 100644
--- a/TESTING/MATGEN/clarge.f
+++ b/TESTING/MATGEN/clarge.f
@@ -113,7 +113,7 @@
       COMPLEX            TAU, WA, WB
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGEMV, CGERC, CLARNV, CSCAL, XERBLA
+      EXTERNAL           CGEMV, CGERC, CLARNV, CSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX, REAL
@@ -123,6 +123,8 @@
       EXTERNAL           SCNRM2
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/claror.f b/TESTING/MATGEN/claror.f
index 6870c321a..95c95c2ca 100644
--- a/TESTING/MATGEN/claror.f
+++ b/TESTING/MATGEN/claror.f
@@ -194,12 +194,14 @@
       EXTERNAL           LSAME, SCNRM2, CLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CGEMV, CGERC, CLACGV, CLASET, CSCAL, XERBLA
+      EXTERNAL           CGEMV, CGERC, CLACGV, CLASET, CSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, CMPLX, CONJG
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       INFO = 0
       IF( N.EQ.0 .OR. M.EQ.0 )
diff --git a/TESTING/MATGEN/clarot.f b/TESTING/MATGEN/clarot.f
index b899c8274..104d68432 100644
--- a/TESTING/MATGEN/clarot.f
+++ b/TESTING/MATGEN/clarot.f
@@ -253,12 +253,14 @@
       COMPLEX            XT( 2 ), YT( 2 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          CONJG
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Set up indices, arrays for ends
 *
diff --git a/TESTING/MATGEN/clatm1.f b/TESTING/MATGEN/clatm1.f
index 48d49897b..1b084ce57 100644
--- a/TESTING/MATGEN/clatm1.f
+++ b/TESTING/MATGEN/clatm1.f
@@ -168,12 +168,14 @@
       EXTERNAL           SLARAN, CLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CLARNV, XERBLA
+      EXTERNAL           CLARNV, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, EXP, LOG, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Decode and Test the input parameters. Initialize flags & seed.
 *
diff --git a/TESTING/MATGEN/clatmr.f b/TESTING/MATGEN/clatmr.f
index e80c4a514..53f825c67 100644
--- a/TESTING/MATGEN/clatmr.f
+++ b/TESTING/MATGEN/clatmr.f
@@ -537,12 +537,14 @@
      $                   CLATM2, CLATM3
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           CLATM1, CSSCAL, XERBLA
+      EXTERNAL           CLATM1, CSSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, CONJG, MAX, MIN, MOD, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Decode and Test the input parameters.
 *             Initialize flags & seed.
diff --git a/TESTING/MATGEN/dlagge.f b/TESTING/MATGEN/dlagge.f
index 44b1c25b4..e5b47b8b6 100644
--- a/TESTING/MATGEN/dlagge.f
+++ b/TESTING/MATGEN/dlagge.f
@@ -137,7 +137,7 @@
       DOUBLE PRECISION   TAU, WA, WB, WN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DGEMV, DGER, DLARNV, DSCAL, XERBLA
+      EXTERNAL           DGEMV, DGER, DLARNV, DSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, MIN, SIGN
@@ -147,6 +147,8 @@
       EXTERNAL           DNRM2
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/dlahilb.f b/TESTING/MATGEN/dlahilb.f
index 5d0351f46..0c7787140 100644
--- a/TESTING/MATGEN/dlahilb.f
+++ b/TESTING/MATGEN/dlahilb.f
@@ -150,13 +150,15 @@
       PARAMETER (NMAX_EXACT = 6, NMAX_APPROX = 11)
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL XERBLA
+      EXTERNAL XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. External Functions
       EXTERNAL DLASET
       INTRINSIC DBLE
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/dlarge.f b/TESTING/MATGEN/dlarge.f
index fc8bcc4be..4608cb5f3 100644
--- a/TESTING/MATGEN/dlarge.f
+++ b/TESTING/MATGEN/dlarge.f
@@ -111,7 +111,7 @@
       DOUBLE PRECISION   TAU, WA, WB, WN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DGEMV, DGER, DLARNV, DSCAL, XERBLA
+      EXTERNAL           DGEMV, DGER, DLARNV, DSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, SIGN
@@ -121,6 +121,8 @@
       EXTERNAL           DNRM2
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/dlaror.f b/TESTING/MATGEN/dlaror.f
index a48129d25..4729d2f0c 100644
--- a/TESTING/MATGEN/dlaror.f
+++ b/TESTING/MATGEN/dlaror.f
@@ -177,12 +177,14 @@
       EXTERNAL           LSAME, DLARND, DNRM2
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DGEMV, DGER, DLASET, DSCAL, XERBLA
+      EXTERNAL           DGEMV, DGER, DLASET, DSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, SIGN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       INFO = 0
       IF( N.EQ.0 .OR. M.EQ.0 )
diff --git a/TESTING/MATGEN/dlarot.f b/TESTING/MATGEN/dlarot.f
index 6655fa481..e7ebd7591 100644
--- a/TESTING/MATGEN/dlarot.f
+++ b/TESTING/MATGEN/dlarot.f
@@ -249,9 +249,11 @@
       DOUBLE PRECISION   XT( 2 ), YT( 2 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DROT, XERBLA
+      EXTERNAL           DROT, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Set up indices, arrays for ends
 *
diff --git a/TESTING/MATGEN/dlatm1.f b/TESTING/MATGEN/dlatm1.f
index cfb144b10..3c8f6f03b 100644
--- a/TESTING/MATGEN/dlatm1.f
+++ b/TESTING/MATGEN/dlatm1.f
@@ -166,12 +166,14 @@
       EXTERNAL           DLARAN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLARNV, XERBLA
+      EXTERNAL           DLARNV, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, EXP, LOG
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Decode and Test the input parameters. Initialize flags & seed.
 *
diff --git a/TESTING/MATGEN/dlatm7.f b/TESTING/MATGEN/dlatm7.f
index 70c2f98ba..bb230b850 100644
--- a/TESTING/MATGEN/dlatm7.f
+++ b/TESTING/MATGEN/dlatm7.f
@@ -155,12 +155,14 @@
       EXTERNAL           DLARAN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLARNV, XERBLA
+      EXTERNAL           DLARNV, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, EXP, LOG
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Decode and Test the input parameters. Initialize flags & seed.
 *
diff --git a/TESTING/MATGEN/dlatmr.f b/TESTING/MATGEN/dlatmr.f
index a914481f7..92d964567 100644
--- a/TESTING/MATGEN/dlatmr.f
+++ b/TESTING/MATGEN/dlatmr.f
@@ -512,12 +512,14 @@
      $                   DLATM2, DLATM3
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLATM1, DSCAL, XERBLA
+      EXTERNAL           DLATM1, DSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX, MIN, MOD
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Decode and Test the input parameters.
 *             Initialize flags & seed.
diff --git a/TESTING/MATGEN/slagge.f b/TESTING/MATGEN/slagge.f
index 626039aa0..087b8af3a 100644
--- a/TESTING/MATGEN/slagge.f
+++ b/TESTING/MATGEN/slagge.f
@@ -137,7 +137,7 @@
       REAL               TAU, WA, WB, WN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SGEMV, SGER, SLARNV, SSCAL, XERBLA
+      EXTERNAL           SGEMV, SGER, SLARNV, SSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, MIN, SIGN
@@ -147,6 +147,8 @@
       EXTERNAL           SNRM2
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/slahilb.f b/TESTING/MATGEN/slahilb.f
index cb92db6db..3bebe1831 100644
--- a/TESTING/MATGEN/slahilb.f
+++ b/TESTING/MATGEN/slahilb.f
@@ -150,13 +150,15 @@
       PARAMETER (NMAX_EXACT = 6, NMAX_APPROX = 11)
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL XERBLA
+      EXTERNAL XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. External Functions
       EXTERNAL SLASET
       INTRINSIC REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/slarge.f b/TESTING/MATGEN/slarge.f
index 392948996..64e6fba5e 100644
--- a/TESTING/MATGEN/slarge.f
+++ b/TESTING/MATGEN/slarge.f
@@ -111,7 +111,7 @@
       REAL               TAU, WA, WB, WN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SGEMV, SGER, SLARNV, SSCAL, XERBLA
+      EXTERNAL           SGEMV, SGER, SLARNV, SSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          MAX, SIGN
@@ -121,6 +121,8 @@
       EXTERNAL           SNRM2
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/slaror.f b/TESTING/MATGEN/slaror.f
index d58902110..0dd0ad083 100644
--- a/TESTING/MATGEN/slaror.f
+++ b/TESTING/MATGEN/slaror.f
@@ -177,12 +177,14 @@
       EXTERNAL           LSAME, SLARND, SNRM2
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SGEMV, SGER, SLASET, SSCAL, XERBLA
+      EXTERNAL           SGEMV, SGER, SLASET, SSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, SIGN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       INFO = 0
       IF( N.EQ.0 .OR. M.EQ.0 )
diff --git a/TESTING/MATGEN/slarot.f b/TESTING/MATGEN/slarot.f
index ed478cb21..ed1ff87eb 100644
--- a/TESTING/MATGEN/slarot.f
+++ b/TESTING/MATGEN/slarot.f
@@ -249,9 +249,11 @@
       REAL               XT( 2 ), YT( 2 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SROT, XERBLA
+      EXTERNAL           SROT, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Set up indices, arrays for ends
 *
diff --git a/TESTING/MATGEN/slatm1.f b/TESTING/MATGEN/slatm1.f
index 8f9e16431..9e38e2892 100644
--- a/TESTING/MATGEN/slatm1.f
+++ b/TESTING/MATGEN/slatm1.f
@@ -166,12 +166,14 @@
       EXTERNAL           SLARAN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SLARNV, XERBLA
+      EXTERNAL           SLARNV, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, EXP, LOG, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Decode and Test the input parameters. Initialize flags & seed.
 *
diff --git a/TESTING/MATGEN/slatm7.f b/TESTING/MATGEN/slatm7.f
index 098d1c2b7..b9ae5d954 100644
--- a/TESTING/MATGEN/slatm7.f
+++ b/TESTING/MATGEN/slatm7.f
@@ -155,12 +155,14 @@
       EXTERNAL           SLARAN
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SLARNV, XERBLA
+      EXTERNAL           SLARNV, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, EXP, LOG, REAL
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Decode and Test the input parameters. Initialize flags & seed.
 *
diff --git a/TESTING/MATGEN/slatmr.f b/TESTING/MATGEN/slatmr.f
index c2cedd21c..3d018db48 100644
--- a/TESTING/MATGEN/slatmr.f
+++ b/TESTING/MATGEN/slatmr.f
@@ -512,12 +512,14 @@
      $                   SLATM2, SLATM3
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           SLATM1, SSCAL, XERBLA
+      EXTERNAL           SLATM1, SSCAL, XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, MAX, MIN, MOD
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Decode and Test the input parameters.
 *             Initialize flags & seed.
diff --git a/TESTING/MATGEN/zlagge.f b/TESTING/MATGEN/zlagge.f
index c9d2bc240..0506acdf9 100644
--- a/TESTING/MATGEN/zlagge.f
+++ b/TESTING/MATGEN/zlagge.f
@@ -141,7 +141,7 @@
       COMPLEX*16         TAU, WA, WB
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGEMV, ZGERC, ZLACGV, ZLARNV, ZSCAL
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGEMV, ZGERC, ZLACGV, ZLARNV, ZSCAL
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, MAX, MIN
@@ -151,6 +151,8 @@
       EXTERNAL           DZNRM2
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/zlaghe.f b/TESTING/MATGEN/zlaghe.f
index 506035061..d449e0827 100644
--- a/TESTING/MATGEN/zlaghe.f
+++ b/TESTING/MATGEN/zlaghe.f
@@ -130,7 +130,7 @@
       COMPLEX*16         ALPHA, TAU, WA, WB
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZAXPY, ZGEMV, ZGERC, ZHEMV, ZHER2,
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZAXPY, ZGEMV, ZGERC, ZHEMV, ZHER2,
      $                   ZLARNV, ZSCAL
 *     ..
 *     .. External Functions ..
@@ -142,6 +142,8 @@
       INTRINSIC          ABS, DBLE, DCONJG, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/zlagsy.f b/TESTING/MATGEN/zlagsy.f
index e030a5f18..4f705182a 100644
--- a/TESTING/MATGEN/zlagsy.f
+++ b/TESTING/MATGEN/zlagsy.f
@@ -130,7 +130,7 @@
       COMPLEX*16         ALPHA, TAU, WA, WB
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZAXPY, ZGEMV, ZGERC, ZLACGV, ZLARNV,
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZAXPY, ZGEMV, ZGERC, ZLACGV, ZLARNV,
      $                   ZSCAL, ZSYMV
 *     ..
 *     .. External Functions ..
@@ -142,6 +142,8 @@
       INTRINSIC          ABS, DBLE, MAX
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/zlahilb.f b/TESTING/MATGEN/zlahilb.f
index e5a317821..a56d918df 100644
--- a/TESTING/MATGEN/zlahilb.f
+++ b/TESTING/MATGEN/zlahilb.f
@@ -175,7 +175,7 @@
      $     (-.5,.5),(.5,.5),(.5,-.5)/
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL XERBLA
+      EXTERNAL XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. External Functions
       EXTERNAL ZLASET, LSAMEN
@@ -183,6 +183,8 @@
       LOGICAL LSAMEN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
       C2 = PATH( 2: 3 )
 *
 *     Test the input arguments
diff --git a/TESTING/MATGEN/zlarge.f b/TESTING/MATGEN/zlarge.f
index 4e1de110e..e20ba8a07 100644
--- a/TESTING/MATGEN/zlarge.f
+++ b/TESTING/MATGEN/zlarge.f
@@ -113,7 +113,7 @@
       COMPLEX*16         TAU, WA, WB
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGEMV, ZGERC, ZLARNV, ZSCAL
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGEMV, ZGERC, ZLARNV, ZSCAL
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, MAX
@@ -123,6 +123,8 @@
       EXTERNAL           DZNRM2
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Test the input arguments
 *
diff --git a/TESTING/MATGEN/zlaror.f b/TESTING/MATGEN/zlaror.f
index b8c4fdfa5..3cc18f5fd 100644
--- a/TESTING/MATGEN/zlaror.f
+++ b/TESTING/MATGEN/zlaror.f
@@ -194,12 +194,14 @@
       EXTERNAL           LSAME, DZNRM2, ZLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZGEMV, ZGERC, ZLACGV, ZLASET, ZSCAL
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZGEMV, ZGERC, ZLACGV, ZLASET, ZSCAL
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DCMPLX, DCONJG
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
       INFO = 0
       IF( N.EQ.0 .OR. M.EQ.0 )
diff --git a/TESTING/MATGEN/zlarot.f b/TESTING/MATGEN/zlarot.f
index ebb5d98f9..4b9373b0d 100644
--- a/TESTING/MATGEN/zlarot.f
+++ b/TESTING/MATGEN/zlarot.f
@@ -253,12 +253,14 @@
       COMPLEX*16         XT( 2 ), YT( 2 )
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          DCONJG
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Set up indices, arrays for ends
 *
diff --git a/TESTING/MATGEN/zlatm1.f b/TESTING/MATGEN/zlatm1.f
index a829d9119..621431bc3 100644
--- a/TESTING/MATGEN/zlatm1.f
+++ b/TESTING/MATGEN/zlatm1.f
@@ -168,12 +168,14 @@
       EXTERNAL           DLARAN, ZLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZLARNV
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZLARNV
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, EXP, LOG
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     Decode and Test the input parameters. Initialize flags & seed.
 *
diff --git a/TESTING/MATGEN/zlatme.f b/TESTING/MATGEN/zlatme.f
index fee1eca6e..6a95731a5 100644
--- a/TESTING/MATGEN/zlatme.f
+++ b/TESTING/MATGEN/zlatme.f
@@ -347,7 +347,7 @@
       EXTERNAL           LSAME, ZLANGE, ZLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLATM1, XERBLA, ZCOPY, ZDSCAL, ZGEMV, ZGERC,
+      EXTERNAL           DLATM1, XERBLA, MKL_INIT_XERBLA, ZCOPY, ZDSCAL, ZGEMV, ZGERC,
      $                   ZLACGV, ZLARFG, ZLARGE, ZLARNV, ZLASET, ZLATM1,
      $                   ZSCAL
 *     ..
@@ -355,6 +355,8 @@
       INTRINSIC          ABS, DCONJG, MAX, MOD
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Decode and Test the input parameters.
 *             Initialize flags & seed.
diff --git a/TESTING/MATGEN/zlatmr.f b/TESTING/MATGEN/zlatmr.f
index 56285e1f4..586b3b227 100644
--- a/TESTING/MATGEN/zlatmr.f
+++ b/TESTING/MATGEN/zlatmr.f
@@ -537,12 +537,14 @@
      $                   ZLATM2, ZLATM3
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           XERBLA, ZDSCAL, ZLATM1
+      EXTERNAL           XERBLA, MKL_INIT_XERBLA, ZDSCAL, ZLATM1
 *     ..
 *     .. Intrinsic Functions ..
       INTRINSIC          ABS, DBLE, DCONJG, MAX, MIN, MOD
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Decode and Test the input parameters.
 *             Initialize flags & seed.
diff --git a/TESTING/MATGEN/zlatms.f b/TESTING/MATGEN/zlatms.f
index 9bad54860..20b750602 100644
--- a/TESTING/MATGEN/zlatms.f
+++ b/TESTING/MATGEN/zlatms.f
@@ -377,7 +377,7 @@
       EXTERNAL           LSAME, DLARND, ZLARND
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLATM1, DSCAL, XERBLA, ZLAGGE, ZLAGHE, ZLAGSY,
+      EXTERNAL           DLATM1, DSCAL, XERBLA, MKL_INIT_XERBLA, ZLAGGE, ZLAGHE, ZLAGSY,
      $                   ZLAROT, ZLARTG, ZLASET
 *     ..
 *     .. Intrinsic Functions ..
@@ -385,6 +385,8 @@
      $                   SIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Decode and Test the input parameters.
 *             Initialize flags & seed.
diff --git a/TESTING/MATGEN/zlatmt.f b/TESTING/MATGEN/zlatmt.f
index 257c2627e..b3f4360a2 100644
--- a/TESTING/MATGEN/zlatmt.f
+++ b/TESTING/MATGEN/zlatmt.f
@@ -385,7 +385,7 @@
       EXTERNAL           ZLARND, DLARND, LSAME
 *     ..
 *     .. External Subroutines ..
-      EXTERNAL           DLATM7, DSCAL, XERBLA, ZLAGGE, ZLAGHE,
+      EXTERNAL           DLATM7, DSCAL, XERBLA, MKL_INIT_XERBLA, ZLAGGE, ZLAGHE,
      $                   ZLAGSY, ZLAROT, ZLARTG, ZLASET
 *     ..
 *     .. Intrinsic Functions ..
@@ -393,6 +393,8 @@
      $                   SIN
 *     ..
 *     .. Executable Statements ..
+*
+      CALL MKL_INIT_XERBLA()
 *
 *     1)      Decode and Test the input parameters.
 *             Initialize flags & seed.
